assets/js/blocks/product-categories/block.js

```diff
@@ -115,43 +100,26 @@ class ProductCategoriesBlock extends Component {
 										{ this.renderOptions( categories ) }
 									</select>
 								</div>
-								<IconButton
+								<button
+									type="button"
+									className="wc-block-product-categories__button"
+									aria-label={ __( 'Go to category', 'woo-gutenberg-products-block' ) }
 									icon="arrow-right-alt2"
-									label={ __( 'Go to category', 'woo-gutenberg-products-block' ) }
 									onClick={ this.onNavigate }
-								/>
+								>
+									<svg aria-hidden="true" role="img" focusable="false" className="dashicon dashicons-arrow-right-alt2" xmlns="http://www.w3.org/2000/svg" width="20" height="20" viewBox="0 0 20 20">
+										<path d="M6 15l5-5-5-5 1-2 7 7-7 7z"></path>
+									</svg>
+								</button>
 							</Fragment>
 						) : (
 							this.renderList( categories )
 						) }
 					</div>
-				) : (
-					<Placeholder
-						className="wc-block-product-categories"
-						icon={ <IconFolder /> }
-						label={ __( 'Product Categories List', 'woo-gutenberg-products-block' ) }
-					>
-						{ __( "This block shows product categories for your store. In order to preview this you'll first need to create a product and assign it to a category.", 'woo-gutenberg-products-block' ) }
-					</Placeholder>
 				) }
 			</Fragment>
 		);
 	}
 }
 
-ProductCategoriesBlock.propTypes = {
-	/**
-	 * The attributes for this block.
-	 */
-	attributes: PropTypes.object.isRequired,
-	/**
-	 * A unique ID for identifying the label for the select dropdown.
-	 */
-	instanceId: PropTypes.number,
-	/**
-	 * Whether this is the block preview or frontend display.
-	 */
-	isPreview: PropTypes.bool,
-};
-
-export default withInstanceId( ProductCategoriesBlock );
+export default withComponentID( ProductCategoriesBlock );
```

Comment:
I know that's very nitpicky, but what do you think about renaming `withComponentID` and `componentID` to `withComponentId` and `componentId` (notice the camel case)? This way it's consistent with `withInstanceId`and `instanceId`.

This also seems to be the consensus they arrived in Gutenberg: https://github.com/WordPress/gutenberg/issues/6741.